File: sample/src/main/java/com/antonionicolaspina/textimageview/sample/MainActivity.java
package com.antonionicolaspina.textimageview.sample;

import android.graphics.PointF;
import android.graphics.Typeface;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.antonionicolaspina.textimageview.TextImageView;

public class MainActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    final TextImageView textImageView = (TextImageView) findViewById(R.id.text_image);
    textImageView.setTypeface(Typeface.DEFAULT_BOLD);

    final TextView textView = (TextView) findViewById(R.id.text);

    textImageView.setOnTextMovedListener(new TextImageView.OnTextMovedListener() {
      @Override
      public void textMoved(PointF position) {
        textView.setText(String.format("Position: [%.2f%%, %.2f%%]", position.x * 100f, position.y * 100f));
      }
    });
  }
}

File: sample/src/main/java/com/antonionicolaspina/textimageview/sample/MainActivity.kt
package com.antonionicolaspina.textimageview.sample

import android.graphics.Color
import android.graphics.Typeface
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.antonionicolaspina.textimageview.Text
import com.antonionicolaspina.textimageview.TextImageView
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    text_image.listener = object : TextImageView.Listener {
      override fun textsChanged(texts: List<Text>) {
        position_textview.text = texts.joinToString("\n")
      }

      override fun textTapped(text: Text) {
        Toast.makeText(this@MainActivity, "Tapped ${text.text}", Toast.LENGTH_SHORT).show()
      }
    }

    add_more_button.setOnClickListener {
      text_image.addText("Text ${Date()}")
      text_image.setTypeface(Typeface.DEFAULT_BOLD)
      text_image.setTextColor(Color.RED)
      text_image.addDropShadow()
    }
  }
}
